File: pharmacy-v1/pharmacy-backend-microservice/blog-service/src/main/java/com/pharmacy_backend/blog_service/service/BlogService.java
package com.pharmacy_backend.blog_service.service;

import com.pharmacy_backend.blog_service.dto.request.BlogRequest;
import com.pharmacy_backend.blog_service.dto.response.BlogResponse;
import com.pharmacy_backend.common.dto.response.ApiResponse;
import com.pharmacy_backend.common.dto.response.PageResponse;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface BlogService {
    ApiResponse<PageResponse<List<BlogResponse>>> getAllBlogs(int pageIndex, int pageSize, String title, String category);
    ApiResponse<BlogResponse> getBlogBySlug(String slug);
    ApiResponse<BlogResponse> getBlogById(Long id);
    ApiResponse<BlogResponse> createBlog(BlogRequest request, MultipartFile thumbnail);
    ApiResponse<BlogResponse> updateBlog(Long id, BlogRequest request, MultipartFile thumbnail);
    ApiResponse<Void> deleteBlog(Long id);
}

File: pharmacy-v1/fe/pharmacy-app/src/api/blogService.ts
import axiosClient from './axiosClient';
import { ApiResponse, PageResponse } from '../types';
import { Blog } from '../types/blog';

const blogService = {
  getAllBlogs: (pageIndex: number = 1, pageSize: number = 10, title?: string, categorySlug?: string) => {
    const params: any = { pageIndex, pageSize };
    if (title) params.title = title;
    if (categorySlug) params.category = categorySlug; 

    return axiosClient.get<ApiResponse<PageResponse<Blog[]>>>('/blogs', { params });
  },

  getBlogBySlug: (slug: string) => {
    return axiosClient.get<ApiResponse<Blog>>(`/blogs/slug/${slug}`);
  }
};

export default blogService;
